esp/clients/ws_dfsclient/ws_dfsclient.cpp

```diff
@@ -330,6 +330,10 @@ class CServiceDistributedFileBase : public CSimpleInterfaceOf<INTERFACE>
     {
         UNIMPLEMENTED_X("CServiceDistributedFileBase::enqueueReplicate");
     }
+    virtual IUserDescriptor *queryUser() const override
+    {
+        return legacyDFSFile->queryUser();
```

Comment:
I may not be following the code correctly, but I don't think this has the right information to be able to implement this.
legacyDFSFile is a member of the base.  It only seems to be intialised by CServiceSuperDistributedFile, by a call that does not provide the user.
I think it could be accessed from dfsFile->queryUserDescriptor(), but that returns a linked pointer.